services-worker/src/main/java/edu/unc/lib/dl/cdr/services/worker/WorkerDaemon.java

```diff
@@ -31,26 +33,53 @@ public void init(DaemonContext context) throws DaemonInitException, Exception {
 
 	@Override
 	public void start() throws Exception {
+		long backoffDelay = 10000;
+		int backoffAttempts = 1;
+		int maxBackoffAttempts = 5;
+
 		LOG.info("Starting the services worker daemon");
 
-		if (appContext == null) {
-			appContext = new ClassPathXmlApplicationContext(new String[] { "service-context.xml" });
-			appContext.registerShutdownHook();
-		} else {
-			appContext.refresh();
-		}
-		
-		Map<String, WorkerPool> workerPools = appContext.getBeansOfType(WorkerPool.class);
-		for (WorkerPool workerPool : workerPools.values()) {
-			workerPool.getWorkerEventEmitter().addListener(this);
-			workerPool.run();
+		while (backoffAttempts <= maxBackoffAttempts) {
+			if (appContext == null) {
+				appContext = new ClassPathXmlApplicationContext(new String[] { "service-context.xml" });
+				appContext.registerShutdownHook();
+			} else {
+				appContext.refresh();
+			}
+
+			ManagementClient managementClient = (ManagementClient) appContext.getBean("managementClient");
+
+			if (managementClient.isRepositoryAvailable()) {
+				Map<String, WorkerPool> workerPools = appContext.getBeansOfType(WorkerPool.class);
+				for (WorkerPool workerPool : workerPools.values()) {
+					workerPool.getWorkerEventEmitter().addListener(this);
+					workerPool.run();
+				}
+
+				break;
+			} else {
+				LOG.warn("Unable to connect to fedora. Retrying starting Worker Daemon. "
+					+ "Retry attempt " + backoffAttempts);
+
+				try {
+					Thread.sleep(backoffDelay * backoffAttempts);
+				} catch (InterruptedException e) {
+					return;
+				}
+
+				if (backoffAttempts == 5) {
```

Comment:
Use the constant for max number of attempts

Reply:
Also, if you put a return in here then you wouldn't need to repeat the cut off in the while loop, and could do a `while (true)`